### PR TITLE
perfect id column match with text possible

### LINKED ISSUE #4206: Text columns should be able to to be identified as a perfect match for an ID column

### PR DESCRIPTION
**Pull request by simha104**
### Pull Request Description
Perfect id column match with text possible
Resolves #4206

### CODE REVIEW

**Comment by machineFL** [comment]
No linked issues found. Please add the corresponding issues in the pull request description. <br/> [Use GitHub automation to close the issue when a PR is merged](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

File: evalml/data_checks/id_columns_data_check.py [hunk 1]
```diff
@@ -180,16 +181,19 @@ def validate(self, X, y=None):
         ]  # columns whose name is "id"
         id_cols = {col: 0.95 for col in cols_named_id}
 
-        for dtypes in [["Double"], ["Integer", "IntegerNullable", "Categorical"]]:
-            X_temp = X.ww.select(include=dtypes)
+        for types in [
+            ["Double"],
+            ["Integer", "IntegerNullable", "Categorical", "Text", "Unknown"],
```
Comment (simha104) [inline evalml/data_checks/id_columns_data_check.py:186]: Integrated with the UI, Text columns are `Text` type, whereas within evalml, Text columns are `Unknown` type. I'm not sure how text columns become `Text` type in the UI when there is no `Text` logical type in Woodwork

Comment (bchen1116) [reply evalml/data_checks/id_columns_data_check.py:186]: @jeff-hernandez any chance you'd know why it's coming in as `Text`?

File: evalml/tests/data_checks_tests/test_id_columns_data_check.py [hunk 2]
```diff
@@ -335,13 +335,15 @@ def test_unidentified_first_col_primary_key(
...
-    if input_type == "integer":
+    if input_type == "Integer":
         X.at[0, "col_1"] = 0
-    elif input_type == "integer_nullable":
+    elif input_type == "IntegerNullable":
         X.at[0, "col_1"] = 0
-    elif input_type == "double":
+    elif input_type == "Double":
         X.at[0, "col_1"] = 0.0
-    elif input_type == "string":
```
Comment (bchen1116) [inline evalml/tests/data_checks_tests/test_id_columns_data_check.py:344]: Is `Text` replacing `string` now? Should we also do an IF for `Unknown`?

Comment (simha104) [reply evalml/tests/data_checks_tests/test_id_columns_data_check.py:344]: The variable `input_type` has no bearing on the type of the first column. That is determined within the block of code within the IF statement. I just changed the name because it was consistent with the rest of the names as being logical types. I could change the name to `Unknown` actually because that is what the logical type of the first column of the data within that IF block is. There is no `Text` Woodwork logical type so I cannot manually set the type of the first column to be that. However, the UI somehow is able to make the logical type of the first column as `Text`

Comment (bchen1116) [reply evalml/tests/data_checks_tests/test_id_columns_data_check.py:344]: Got it, that makes sense!

File: docs/source/release_notes.rst [hunk 3]
```diff
@@ -3,6 +3,7 @@ Release Notes
 **Future Releases**
     * Enhancements
     * Fixes
+        * Text columns can be identified as a perfect match as an id column :pr:`4203`
```
Comment (bchen1116) [inline docs/source/release_notes.rst:6]: Might want to reword this to clarify, maybe something like "``IDColumnDataCheck`` now works with Unknown data type" or something

**Review by bchen1116 — APPROVED** [review]
LGTM! Just left one comment for clarity

File: docs/source/release_notes.rst [hunk 4]
```diff
@@ -3,6 +3,7 @@ Release Notes
 **Future Releases**
     * Enhancements
     * Fixes
+        * `IDColumnDataCheck` now works with Unknown data type :pr:`4203`
```
Comment (ParthivNaresh) [inline docs/source/release_notes.rst:6]: Nit: IDColumnsDataCheck

**Review by ParthivNaresh — APPROVED** [review]
Looks good!
